### PR TITLE
Feature - remove all resource group locks

### LINKED ISSUE #41: Provide script to remove (all) locks on a resource group
**Is your feature request related to a problem? Please describe.**
In some deployments resource-locks are being assigned. To help in removing these quickly, a script could be provided.
While this seems dangerous, only users with sufficient access rights are allowed to delete locks.

**Describe the solution you'd like**
```
Remove-ResourceGroupLocks
 -ResourceGroupName (required)
 -LockName (optional)
```

**Additional context**
Can be based on this script:
https://github.com/mbraekman/Powershell/blob/master/Azure/Remove%20All%20Locks%20In%20ResourceGroup.ps1

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds a PS module function to remove all resource locks in a resource group on Azure.

Closes #41

### CODE REVIEW

File: docs/preview/features/powershell/arm.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ layout: default
 
 This module provides the following capabilities:
 - [Injecting content into an ARM template](#injecting-content-into-an-arm-template)
+- [Removing resource locks on azure](#removing-resource-locks-on-azure)
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:10]:
```suggestion
- [Removing resource locks within an Azure Resource Group](#removing-resource-locks-within-an-azure-resource-group)
```

File: docs/preview/features/powershell/arm.md [hunk 2]
```diff
@@ -56,3 +57,34 @@ PS> Inject-ArmContent -Path deploy\arm-template.json
 
 **Recommendations**
 Always inject the content in your ARM template as soon as possible, preferably during release build that creates the artifact
+
+## Removing resource locks on Azure
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:61]:
```suggestion
## Removing resource locks within an Azure Resource Group
```

File: docs/preview/features/powershell/arm.md [hunk 3]
```diff
@@ -56,3 +57,34 @@ PS> Inject-ArmContent -Path deploy\arm-template.json
...
+
+In some deployments resource-locks are being assigned. To help in removing these quickly, we have provided you with a function that removes all the existing locks on the resource group.
+
+While this seems dangerous, only users with sufficient access rights are allowed to delete locks.
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:65]: Perhaps include some information regarding the specific access rights required:
```suggestion
While this seems dangerous, only users with access to the following actions are allowed to delete locks:
 - _Microsoft.Authorization/*_
 - _Microsoft.Authorization/locks/*_
 
 Of the built-in roles, only **Owner** and **Access Administrator** are granted those actions.
```

File: docs/preview/features/powershell/arm.md [hunk 4]
```diff
@@ -56,3 +57,34 @@ PS> Inject-ArmContent -Path deploy\arm-template.json
...
+
+| Parameter           | Mandatory | Description                                                                                       |
+| ------------------- | --------- | ------------------------------------------------------------------------------------------------- |
+| `ResourceGroupName` | yes       | The name of the resource group where the locks should be removed                                  |
+| `LockName`          | no        | The optional name of the lock to remove. When this is not provided, all the locks will be removed |
+
+**Usage*
+
+When you want to remove all the resource locks, no matter what the name or the sub-location:
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:74]:
```suggestion
When you want to remove all the resource locks within a specific resource group, no matter what the name or the sub-location:
```

File: docs/preview/features/powershell/arm.md [hunk 5]
```diff
@@ -56,3 +57,34 @@ PS> Inject-ArmContent -Path deploy\arm-template.json
...
+
+```powershell
+PS> Remove-AzResourceGroupLocks -ResourceGroupName "your-resource-group-name"
+# Retrieving all locks in resourceGroup 'your-resource-group-name'
+# Start removing all locks in resourceGroup 'your-resource-group-name'
+# All locks in resourceGroup 'your-resource-group-name' have been removed
+```
+
+When you want to remove a specific resource lock, with a given name:
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:83]:
```suggestion
When you want to remove a specific resource lock within a given resource group, with a given name:
```

**Review by mbraekman — COMMENTED** [review]
Some remarks in general, but I would prefer to see this script added into an Arcus.Scripting.**Security** module, instead of adding this into the ARM-module.

File: docs/preview/features/powershell/arm.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/arm.md]: Ok, thank you for this!

**Comment by mbraekman** [comment]
Just 1 more thing to do: moving this out of the ARM-module, into a Security-module 😄

**Comment by stijnmoreels** [comment]
> Just 1 more thing to do: moving this out of the ARM-module, into a Security-module 😄

That's done! ✅

File: docs/preview/features/powershell/arm.md [hunk 6]
```diff
@@ -7,6 +7,7 @@ layout: default
 
 This module provides the following capabilities:
 - [Injecting content into an ARM template](#injecting-content-into-an-arm-template)
+- [Removing resource locks within an Azure Resource Group](#removing-resource-locks-within-an-azure-resource-group)
```
Comment (mbraekman) [inline docs/preview/features/powershell/arm.md:10]: This line can be removed from this document.

File: docs/preview/features/powershell/azure-security.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+---
+title: "Scripts related to Azure security"
+layout: default
+---
+
+# Azure Security
+
+This module provides the following capabilities:
+- [Removing resource locks on azure](#removing-resource-locks-on-azure)
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-security.md:9]:
```suggestion
- [Removing resource locks from an Azure resource group](#removing-resource-locks-from-an-azure-resource-group)
```
So it clearly states it's about removing locks from a specific resource group, not removing locks in general.

File: docs/preview/features/powershell/azure-security.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Installation
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Import-Module -Name Arcus.Scripting.Security
+```
+
+## Removing resource locks on Azure
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-security.md:19]:
```suggestion
## Removing resource locks from an Azure resource group
```

**Review by mbraekman — COMMENTED** [review]
Some small documentation changes, other than that it's looking good!

File: docs/preview/features/powershell/arm.md (hunk 6 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/arm.md]: Woops, thanks!

**Review by mbraekman — APPROVED** [review]

**Review by mbraekman — COMMENTED** [review]
Seems like there is an issue with the unit-tests:

```
Removing the lock with ID: 
[-] Arcus.ARM remove resource group locks.Removes all resource group locks without providing lock name 71ms (69ms|2ms)
 ValidationMetadataException: The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 ParameterBindingValidationException: Cannot validate argument on parameter 'LockId'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 at <ScriptBlock>, /home/vsts/work/1/s/src/Arcus.Scripting.Security/Scripts/Remove-AzResourceGroupLocks.ps1:21
 at Remove-AzResourceGroupLocks, /home/vsts/work/1/s/src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1:20
 at <ScriptBlock>, /home/vsts/work/1/s/src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.Security.tests.ps1:17
```

Could you have a look at this, please?

**Pull request merged**